src/main/java/seedu/address/model/rule/Rule.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.rule;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents a Rule in budget buddy.
+ * Guarantees: predicate and action are present and not null, field values are validated, immutable.
```

Comment:
might want to have some validation methods (stubs are possible) if we're specifying here that RulePredicate is validated.

Reply:
Will consider this. Thanks.